WoofWare.PawPrint/IlMachineState.fs

```diff
@@ -298,17 +336,48 @@ module IlMachineState =
         | TypeDefn.FromReference (ref, _typeKind) ->
             let state, assy, ty = resolveTypeFromRef loggerFactory assy ref genericArgs state
             state, assy, ty
+        | TypeDefn.PrimitiveType prim ->
+            let ty =
+                match prim with
+                | PrimitiveType.Boolean -> corelib.Boolean
+                | PrimitiveType.Char -> corelib.Char
+                | PrimitiveType.SByte -> corelib.SByte
+                | PrimitiveType.Byte -> corelib.Byte
+                | PrimitiveType.Int16 -> corelib.Int16
+                | PrimitiveType.UInt16 -> corelib.UInt16
+                | PrimitiveType.Int32 -> corelib.Int32
+                | PrimitiveType.UInt32 -> corelib.UInt32
+                | PrimitiveType.Int64 -> corelib.Int64
+                | PrimitiveType.UInt64 -> corelib.UInt64
+                | PrimitiveType.Single -> corelib.Single
+                | PrimitiveType.Double -> corelib.Double
+                | PrimitiveType.String -> corelib.String
+                | PrimitiveType.TypedReference -> failwith "todo"
+                | PrimitiveType.IntPtr -> failwith "todo"
+                | PrimitiveType.UIntPtr -> failwith "todo"
+                | PrimitiveType.Object -> failwith "todo"
+                |> TypeInfo.mapGeneric (fun _ -> failwith "none of these types are generic")
+
+            state, corelib.Corelib, ty
+        | TypeDefn.GenericTypeParameter param ->
+            match genericArgs with
+            | None -> failwith "tried to resolve generic parameter without generic args"
+            | Some genericArgs ->
+                let arg = genericArgs.[param]
+                // TODO: this assembly is probably wrong?
```

Comment:
Yeah, this indicates that we should simply not be handing assemblies round all over the place. They should be contained entirely within the TypeDefn.